usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh

```diff
@@ -1,71 +1,87 @@
-# rebuild the initramfs if the drivers changed
-#
-# probably not required, but I prefer to rely on this information when it is backed by udev
-# FIXME: who is 'I'?
-# Perhaps Schlomo Schapiro or someone who made the "P2V patch from Heinlein Support"?
-# (see commit 844d50b75ac4b7722f4fee7a5ee3350b93f3adb7)
-# And what happens if there is no 'have_udev'? Why is everything o.k. then to just 'return 0'?
-have_udev || return 0
 
-# check if we need to do something
-if test -s $TMP_DIR/storage_drivers && ! diff $TMP_DIR/storage_drivers $VAR_DIR/recovery/storage_drivers >&8 ; then
-	# remember, diff returns 0 if the files are the same
+# Rebuild the initramfs if the drivers changed:
 
-	# merge new drivers with previous initrd modules
-	# BUG: we only add modules to the initrd, we don't take old ones out
-	#      could be done better, but might not be worth the risk
-	INITRD_MODULES="$( sort -t " " -k 1 -u $TMP_DIR/storage_drivers $TARGET_FS_ROOT/etc/initramfs-tools/module[s] )"
-	# use [] to skip file if it does not exist
-	# -t " " -k 1 tries to keep the comments unsorted
+# Skip if there is nothing to do.
+# During "rear recover" 260_recovery_storage_drivers.sh creates $TMP_DIR/storage_drivers
+if ! test -s $TMP_DIR/storage_drivers ; then
+    Log "Skip recreating initramfs: No needed storage drivers ('$TMP_DIR/storage_drivers' is empty)"
+    return 0
+fi
+# During "rear mkbackup/mkrescue" 260_storage_drivers.sh creates $VAR_DIR/recovery/storage_drivers
+if cmp -s $TMP_DIR/storage_drivers $VAR_DIR/recovery/storage_drivers ; then
+    Log "Skip recreating initramfs: '$TMP_DIR/storage_drivers' and '$VAR_DIR/recovery/storage_drivers' are same"
```

Comment:
I even asked a native English speaking colleague
and you are right:
It is like "foo and bar are the same"
versus "foo and bar are same as baz".